### DISCUSSION TITLE
Add Galaxy Validation

### DISCUSSION THREAD

**Discussion by minsis** [ansible/ansible-lint#4404] [Ideas]
It would be nice if additional galaxy linting was added to ansible-lint.

As an example in my galaxy.yml under tags I was getting an invalid tag error when pushing my collection.
Another issue I had was a missing readme file for my role.

It would be nice to maybe add in the same ruleset that galaxy/automation hub would require in order to push your collection.

**Comment by minsis** [comment]
An example of the of the tag I had was `agent-client-collector`.

**Comment by sean-m-sullivan** [comment]
I disagree, this is not the kitchen sink, it would be nice,  however this can be run with this
https://github.com/ansible/ansible-content-actions/blob/main/.github/workflows/build_import.yaml

That is the actual process hub/galaxy runs so will give you the exact results.

**Comment by minsis** [comment]
So validate somethings but not others? Doesn't make sense.

ansible-lint already validates some galaxy related things, but missing other validations. If anything the validation for `galaxy.yml` file should be reported as a bug which I'll get opened here shortly.

With that said I still think it's worth discussing adding the rest of the galaxy validations. As an example, if the `meta/runtime.yml` file is missing then the linter fails. So the same concept should be applied to other parts of validation like missing `README.md` files in roles.

**Comment by minsis** [comment]
Created bug report for the `galaxy[tags]` alignment issue: https://github.com/ansible/ansible-lint/issues/4409

I also did some more searching and it looks like I'm not the only one in the same boat with this feature idea:

https://github.com/ansible/ansible-lint/issues/4032
https://github.com/ansible/ansible-lint/issues/2874

At one point someone had a PR to add galaxy-importer to ansible-lint but that was dropped. It also seems that galaxy-importer relies on ansible-lint to do some of the linting as well. So it kind of seems linting is kind of spread out between different libraries making everything inconsistent.
